grails-app/controllers/au/org/ala/ecodata/CommentController.groovy

```diff
@@ -0,0 +1,161 @@
+package au.org.ala.ecodata
+
+import grails.converters.JSON
+import groovy.json.JsonSlurper
+
+class CommentController {
+    def outputService
```

Comment:
Should use types where they are known - it improves readability, and you'll get more assistance from the IDE if you define variables etc with types.